File: AIRMAM5/Scripts/Models/Interface/UserCode/MainSubCodeListModel.ts
import { SelectListItem } from '../../Interface/Shared/ISelectListItem';

/** */
export interface MainSubCodeListModel {
    /**主代碼編號 */
    MainCodeId: string;
    /**主代碼名稱 */
    MainCodeName: string;
    /**子代碼列表 */
    SubCodeList: Array<SelectListItem>;
}

File: AIRMAM5.DBEntity/Models/CodeSet/MainSubCodeListModel.cs
using System.Collections.Generic;
using System.Web.Mvc;

namespace AIRMAM5.DBEntity.Models.CodeSet
{
    /// <summary>
    /// 主代碼帶子代碼清單model
    /// </summary>
    public class MainSubCodeListModel
    {
        public MainSubCodeListModel() { }

        /// <summary>
        /// 主代碼
        /// </summary>
        public string MainCodeId { get; set; } = string.Empty;

        /// <summary>
        /// 主代碼名稱
        /// </summary>
        public string MainCodeName { get; set; } = string.Empty;

        /// <summary>
        /// 子代碼列表 <see cref="SelectListItem"/>
        /// </summary>
        public List<SelectListItem> SubCodeList { get; set; } = new List<SelectListItem>();
    }

}
